cmd/data-node/commands/start/node_pre.go

```diff
@@ -61,122 +58,71 @@ func (l *NodeCommand) persistentPre([]string) (err error) {
-	l.tradesSub = sqlsubscribers.NewTradesSubscriber(l.tradeService, l.Log)
-	l.rewardsSub = sqlsubscribers.NewReward(l.rewardService, l.Log)
-	l.marketCreatedSub = sqlsubscribers.NewMarketCreated(l.marketsService, l.Log)
-	l.marketUpdatedSub = sqlsubscribers.NewMarketUpdated(l.marketsService, l.Log)
-	l.delegationsSub = sqlsubscribers.NewDelegation(l.delegationService, l.Log)
-	l.epochSub = sqlsubscribers.NewEpoch(l.epochService, l.Log)
-	l.depositSub = sqlsubscribers.NewDeposit(l.depositService, l.Log)
-	l.withdrawalSub = sqlsubscribers.NewWithdrawal(l.withdrawalService, l.Log)
-	l.proposalsSub = sqlsubscribers.NewProposal(l.governanceService, l.Log)
-	l.votesSub = sqlsubscribers.NewVote(l.governanceService, l.Log)
-	l.marginLevelsSub = sqlsubscribers.NewMarginLevels(l.riskService, l.accountStore, l.Log)
-	l.riskFactorSub = sqlsubscribers.NewRiskFactor(l.riskFactorService, l.Log)
-	l.netParamSub = sqlsubscribers.NewNetworkParameter(l.networkParameterService, l.Log)
-	l.checkpointSub = sqlsubscribers.NewCheckpoint(l.checkpointService, l.Log)
-	l.positionsSub = sqlsubscribers.NewPosition(l.positionService, l.Log)
-	l.oracleSpecSub = sqlsubscribers.NewOracleSpec(l.oracleSpecService, l.Log)
-	l.oracleDataSub = sqlsubscribers.NewOracleData(l.oracleDataService, l.Log)
-	l.liquidityProvisionSub = sqlsubscribers.NewLiquidityProvision(l.liquidityProvisionService, l.Log)
-	l.transferSub = sqlsubscribers.NewTransfer(l.transfersStore, l.accountService, l.Log)
-	l.stakeLinkingSub = sqlsubscribers.NewStakeLinking(l.stakeLinkingService, l.Log)
-	l.notarySub = sqlsubscribers.NewNotary(l.notaryService, l.Log)
-	l.multiSigSignerEventSub = sqlsubscribers.NewERC20MultiSigSignerEvent(l.multiSigService, l.Log)
-	l.keyRotationsSub = sqlsubscribers.NewKeyRotation(l.keyRotationsService, l.Log)
-	l.ethereumKeyRotationsSub = sqlsubscribers.NewEthereumKeyRotation(l.ethereumKeyRotationsService, l.Log)
-	l.nodeSub = sqlsubscribers.NewNode(l.nodeService, l.Log)
-	l.marketDepthSub = sqlsubscribers.NewMarketDepth(l.marketDepthService)
-}
-
-func (l *NodeCommand) setupStoresSQL() error {
+func (l *NodeCommand) initialiseDatabase() (*sqlstore.ConnectionSource, error) {
 	var err error
 	if l.conf.SQLStore.UseEmbedded {
-		l.embeddedPostgres, err = sqlstore.StartEmbeddedPostgres(l.Log, l.conf.SQLStore,
+		l.embeddedPostgres, _, err = sqlstore.StartEmbeddedPostgres(l.Log, l.conf.SQLStore,
 			l.vegaPaths.StatePathFor(paths.DataNodeStorageHome))
 		if err != nil {
-			return fmt.Errorf("failed to start embedded postgres: %w", err)
+			return nil, fmt.Errorf("failed to start embedded postgres: %w", err)
 		}
 	}
 
+	hasVegaSchema, err := snapshot.HasVegaSchema(l.ctx, l.conf.SQLStore.ConnectionConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to check if database is empty: %w", err)
+	}
+
+	// If it's an empty database, recreate it with correct locale settings
+	if !hasVegaSchema {
+		sqlstore.RecreateVegaDatabase(l.ctx, l.Log, l.conf.SQLStore.ConnectionConfig)
```

Comment:
Should probably handle the error that could be returned  by `RecreateVegaDatabase()`.